studio/src/utils/index.ts

```diff
@@ -216,44 +224,79 @@ export function isSensitiveEnvVar(key: string) {
   if (!key) {
     return false;
   }
-  if (key.includes("APIKEY")) {
-    return true;
-  }
-  if (key.includes("API_KEY")) {
-    return true;
-  }
-  if (key.includes("ACCESS")) {
-    return true;
-  }
-  if (key.includes("AUTH_")) {
-    return true;
-  }
-  if (key.includes("CREDENTIALS")) {
-    return true;
-  }
-  if (key.includes("CERTIFICATE")) {
-    return true;
-  }
-  if (key.includes("PASSPHRASE")) {
-    return true;
-  }
-  if (key.includes("DATABASE_URL")) {
-    return true;
-  }
-  if (key.includes("CONNECTION_STRING")) {
-    return true;
-  }
-  if (key.includes("SECRET")) {
-    return true;
-  }
-  if (key.includes("PASSWORD")) {
-    return true;
+
+  return (
+    key.includes("APIKEY") ||
+    key.includes("API_KEY") ||
+    key.includes("ACCESS") ||
+    key.includes("AUTH_") ||
+    key.includes("CREDENTIALS") ||
+    key.includes("CERTIFICATE") ||
+    key.includes("PASSPHRASE") ||
+    key.includes("DATABASE_URL") ||
+    key.includes("CONNECTION_STRING") ||
+    key.includes("SECRET") ||
+    key.includes("PASSWORD") ||
+    key.includes("PRIVATE") ||
+    key.includes("TOKEN")
+  );
+}
+
+export function constructRequestorBody(bodyValue: string): RequestorBody {
+  try {
+    const parsed = JSON.parse(bodyValue);
+    const result = RequestorBodySchema.safeParse(parsed);
+    if (result.success) {
+      return result.data;
+    }
+  } catch {
+    // swallow error
   }
-  if (key.includes("PRIVATE")) {
-    return true;
+
+  return {
+    type: "text",
+    value: bodyValue,
+  };
+}
+
+export function createKeyValueParametersFromValues(
+  values: Array<{ key: string; value: string }>,
+) {
+  return values.map(({ key, value }) => {
+    return {
+      id: createParameterId(),
+      key,
+      value,
+      enabled: true,
+    };
+  });
+}
+
+export function createObjectFromKeyValueParameters<
```

Comment:
as far as i can tell, this is only used in cases where we sync with a collection - however, one thing to be wary of is that this function ignored whether a parameter is enabled/disabled. which is to say, it will copy/sync parameters that have been disabled by the end user